File: contacting-conf-web/src/main/java/nl/ing/api/contacting/conf/domain/model/surveysetting/SurveySetting.java
package nl.ing.api.contacting.conf.domain.model.surveysetting;

import java.util.Optional;

public record SurveySetting(
        Optional<Long> id,
        long accountId,
        String name,
        String channel,
        String channelDirection,
        String voiceSurveyId,
        Optional<String> callflowName,
        Optional<Integer> minFrequency,
        Optional<Long> delay,
        Optional<Float> surveyOfferRatio,
        Optional<Long> minContactLength,
        boolean surveyForTransfers
) {}

File: contacting-conf-web/src/main/scala/nl/ing/api/contacting/conf/repository/model/SurveySetting.scala
package nl.ing.api.contacting.conf.repository.model

/**
 * @author Ayush Mittal
 */
case class SurveySetting(id: Option[Long],
                         accountId: Long,
                         name: String,
                         channel: String,
                         channelDirection: String,
                         voiceSurveyId: String,
                         callflowName: Option[String],
                         minFrequency: Option[Int],
                         delay: Option[Long],
                         surveyOfferRatio: Option[Float],
                         minContactLength: Option[Long],
                         surveyForTransfers: Boolean)
